### PR TITLE
Setuptools upgrade

### PR DESCRIPTION
**Pull request by RonnyPfannschmidt**
this pr drops the alternative script names and support for setuptools without marker support

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -69,46 +60,32 @@ def main():
...
+            ':sys_platform=="win32"': ['colorama'],
+        },
+        # pluggy is vendored in _pytest.vendored_packages
         packages=['_pytest', '_pytest.assertion', '_pytest._code', '_pytest.vendored_packages'],
         py_modules=['pytest'],
         zip_safe=False,
     )
 
 
-def cmdline_entrypoints(versioninfo, platform, basename):
```
Comment (nicoddemus) [inline setup.py]: Hey @RonnyPfannschmidt, I wonder at which point where those alternative script names necessary, and why that's no longer the case?

File: setup.py [hunk 2]
```diff
@@ -43,22 +44,12 @@ def has_environment_marker_support():
     try:
         return pkg_resources.parse_version(setuptools.__version__) >= pkg_resources.parse_version('0.7.2')
     except Exception as exc:
-        sys.stderr.write("Could not test setuptool's version: %s\n" % exc)
-        return False
+        raise SystemExit("setuptools does not yet support markers: %s\n" % exc)
```
Comment (nicoddemus) [inline setup.py]: I think we should give a more helpful message here: `"setuptools version is %s, which does not support markers. Please update your setuptools version"` or so.

File: setup.py [hunk 3]
```diff
@@ -43,22 +44,12 @@ def has_environment_marker_support():
     try:
         return pkg_resources.parse_version(setuptools.__version__) >= pkg_resources.parse_version('0.7.2')
```
Comment (nicoddemus) [inline setup.py]: This should check if the comparison is False and raise the same `SystemExit` below; returning `False` to the caller has no effect after all.

**Comment by nicoddemus** [comment]
Other than my comments, looks good to me. I wonder if we should mention something in the `CHANGELOG`, although I'm not so sure it is really needed (after all `setuptools` pre-1.0 is really ancient).

**Comment by The-Compiler** [comment]
FWIW Ubuntu Precise (12.04) and Debian Wheezy (oldstable) are both still supported upstream and have a pre-1.0 setuptools version. Probably the same for things like CentOS and RHEL too.

**Comment by nicoddemus** [comment]
Oh I didn't know that. In this case, should we wait to merge this, or can we expect users to have a more up to date version anyway?

**Comment by The-Compiler** [comment]
I really don't know - I guess people on those distributions who want to use pytest (e.g. via pip/virtualenv) will also get that old setuptool in their virtualenvs? If so, that'd be a rather bad compatibility break, and probably too early...

**Comment by RonnyPfannschmidt** [comment]
closing due to valid concerns

**Pull request closed without merging**